packages/lib-classifier/src/store/DrawingStore.js

```diff
@@ -1,56 +1,175 @@
+        const snapshotType = getType(snapshot)
+        if (snapshotType.name === 'LineStore') return LineStore
+        if (snapshotType.name === 'PointStore') return PointStore
+        return undefined
+      }
+    }, LineStore, PointStore)),
+    marks: types.map(types.union({
+      dispatcher: (snapshot) => {
+        const snapshotType = getType(snapshot)
+        if (snapshotType.name === 'LineStore') return LineStore
+        if (snapshotType.name === 'PointStore') return PointStore
+        return undefined
+      }
+    }, LineStore, PointStore))
   })
+  .views(self => ({
+    get activeDrawingTask () {
+      const [task] = getRoot(self).workflowSteps.activeStepTasks
+        .filter(task => task.type === 'drawing')
+      return task
+    },
+    get coordinateStream () {
+      const pointerDownStream = self.eventStream.pipe(
+        filter(event => event.type === 'pointerdown')
+      )
+
+      const coordinateStream = self.eventStream.pipe(
+        skipUntil(pointerDownStream),
+        map(event => self.convertEvent(event))
+      )
+
+      return coordinateStream
+    },
+    get isDrawingInActiveWorkflowStep () {
+      return getRoot(self).workflowSteps.activeStepTasks
+        .some(task => task.type === 'drawing')
+    }
+  }))
   .volatile(self => ({
-    eventStream: new Subject()
+    eventStream: new Subject(),
+    svg: null
   }))
   .actions(self => {
     function afterAttach () {
+      createDrawingTaskObserver()
       createClassificationObserver()
       createWorkflowStepsObserver()
     }
 
+    function createDrawingTaskObserver () {
+      const drawingTaskDisposer = autorun(() => {
+        const validDrawingTaskReference = isValidReference(() => getRoot(self).drawing.activeDrawingTask)
+        if (validDrawingTaskReference) {
+          self.createMark()
+        }
+      }, { name: 'DrawingStore DrawingTask Observer autorun' })
+      addDisposer(self, drawingTaskDisposer)
+    }
+
     function createClassificationObserver () {
       const classificationDisposer = autorun(() => {
-        onAction(getRoot(self).classifications, (call) => {
+        onAction(getRoot(self), (call) => {
           if (call.name === 'completeClassification') self.reset()
         })
-      })
+      }, { name: 'DrawingStore Classification Observer autorun' })
       addDisposer(self, classificationDisposer)
     }
 
     function createWorkflowStepsObserver () {
       const workflowStepsDisposer = autorun(() => {
-        onAction(getRoot(self).workflowSteps, (call) => {
+        onAction(getRoot(self), (call) => {
           if (call.name === 'selectStep') self.reset()
         })
-      })
+      }, { name: 'DrawingStore WorkflowStep Observer autorun' })
 
       addDisposer(self, workflowStepsDisposer)
     }
 
-    function reset () {
-      self.activeDrawingTool = 0
+    function createMark () {
+      const MarkStore = getMarkStore(self.activeDrawingTask.tools[self.activeDrawingToolIndex].type)
```

Comment:
Maybe the drawing task should have an active tool view, so that you can reference `drawingTask.activeTool` here? It feels a bit weird that tools are stored on the drawing task, but the active tool index is stored in this store.

Reply:
Resolved in #1266.